meta/store.go

```diff
@@ -416,6 +424,79 @@ func (s *Store) changeState(state raftState) error {
 	return nil
 }
 
+// monitorPeerHealth periodically checks if we have a node that can be promoted to a
+// raft peer to fill any missing slots.
+// This function runs in a separate goroutine.
+func (s *Store) monitorPeerHealth() {
+	defer s.wg.Done()
+
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		// Wait for next tick or timeout.
+		select {
+		case <-ticker.C:
+		case <-s.closing:
+			return
+		}
+		if err := s.promoteNodeToPeer(); err != nil {
+			s.Logger.Printf("error promoting node to raft peer: %s", err)
+		}
+	}
+}
+
+func (s *Store) promoteNodeToPeer() error {
+	// Only do this if you are the leader
+	if !s.IsLeader() {
+		return nil
+	}
+
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	if s.raftState == nil {
```

Comment:
This may be able to be removed now.  There were places in the code that I had to fix that weren't locking before changing this variable.  I'll see what happens when I remove it.